x-pack/plugins/fleet/server/integration_tests/fleet_setup.test.ts

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import Path from 'path';
+
+import { range } from 'lodash';
+
+import type { ISavedObjectsRepository } from '@kbn/core/server';
+import type { TestElasticsearchUtils, createRoot } from '@kbn/core-test-helpers-kbn-server';
+import { getSupertest, createRootWithCorePlugins } from '@kbn/core-test-helpers-kbn-server';
+
+import type {
+  AgentPolicySOAttributes,
+  Installation,
+  OutputSOAttributes,
+  PackagePolicySOAttributes,
+} from '../types';
+
+type Root = ReturnType<typeof createRoot>;
+
+const startAndWaitForFleetSetup = async (root: Root) => {
+  const start = await root.start();
+
+  const isFleetSetupRunning = async () => {
+    const statusApi = getSupertest(root, 'get', '/api/status');
+    const resp: any = await statusApi.send();
+    const fleetStatus = resp.body?.status?.plugins?.fleet;
+    if (fleetStatus?.meta?.error) {
+      throw new Error(`Setup failed: ${JSON.stringify(fleetStatus)}`);
+    }
+
+    const isRunning = !fleetStatus || fleetStatus?.summary === 'Fleet is setting up';
+    return isRunning;
+  };
+
+  while (await isFleetSetupRunning()) {
+    await new Promise((resolve) => setTimeout(resolve, 2000));
+  }
+
+  return start;
+};
+
+const createAndSetupRoot = async (config?: object, index?: number) => {
+  const root = createRootWithCorePlugins(
+    {
+      xpack: {
+        fleet: config,
+      },
+      logging: {
+        appenders: {
+          file: {
+            type: 'file',
+            fileName: Path.join(__dirname, `logs_${Math.floor(Math.random() * 100)}.log`),
+            layout: {
+              type: 'json',
+            },
+          },
+        },
+        loggers: [
+          {
+            name: 'root',
+            appenders: ['file'],
+          },
+          {
+            name: 'plugins.fleet',
+            appenders: ['file'],
+            level: 'info',
+          },
+        ],
+      },
+    },
+    { oss: false }
+  );
+
+  await root.preboot();
+  await root.setup();
+  return root;
+};
+
+/**
+ * Verifies that multiple Kibana instances running in parallel will not create duplicate preconfiguration objects.
+ */
+describe.skip('Fleet setup preconfiguration with multiple instances Kibana', () => {
```

Comment:
I've used these tests locally to start es and run 2 kibana instances as separate processes.
I've tried to run it as a CI task, but didn't see any output, not sure how to check the result of processes running in the background.

Alternatively we can use [multiple zones in scheduled scale tests](https://github.com/elastic/observability-perf/pull/779) to make sure the bug doesn't surface again. (the test would fail if fleet-server doesn't come online)

Reply:
One issue is that the `jest integration` step is picking up these tests if not skipped, and they fail because the `fleet_setup.test.ts` test requires es to be started separately in `es.test.ts`. I tried moving the files to another folder, but then the `jest_integration` script doesn't find them, and it doesn't seem easy to add a new script.